package step01.basic;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

import org.junit.Test;

public class ConnectionProperty2 {
	
	//select * from dept
	@Test
    public void connect() throws SQLException {
		Connection conn = null;
		Statement stmt = null;
		ResultSet rs = null;
		
        try {
            conn = DBUtil.getConnection();
            stmt = conn.createStatement();
            rs = stmt.executeQuery("select * from dept");
            
        } catch (SQLException e) {
            e.printStackTrace();
        } finally { // 예외 발생 여부와 무관하게 100% 실행, 신뢰영역
            // 자원 해제 필수!
            stmt.close();
            rs.close();
            conn.close();

        }
        
        System.out.println("예외 처리로 인해 정상, 비정상 이어도 실행되는 영역");
    }
	
}